src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -8054,7 +7905,136 @@ VOID MethodTableBuilder::PlaceThreadStaticFields()
+    if (hasNonTrivialParent)
+    {
+        isBlittable &= pParentMT->IsBlittable() == TRUE;
+    }
+
+    bool isAutoLayoutOrHasAutoLayoutField = ((nestedFieldFlags & EEClassLayoutInfo::NestedFieldFlags::AutoLayout) == EEClassLayoutInfo::NestedFieldFlags::AutoLayout);
+    if (hasNonTrivialParent)
+    {
+        isAutoLayoutOrHasAutoLayoutField &= pParentMT->IsAutoLayoutOrHasAutoLayoutField() == TRUE;
+    }
+
+    bool hasInt128Field = (pParentMT && pParentMT->IsInt128OrHasInt128Fields())
+        || ((nestedFieldFlags & EEClassLayoutInfo::NestedFieldFlags::Int128) == EEClassLayoutInfo::NestedFieldFlags::Int128);
+
+    bool isAlign8 = ((nestedFieldFlags & EEClassLayoutInfo::NestedFieldFlags::Align8) == EEClassLayoutInfo::NestedFieldFlags::Align8)
+#ifdef FEATURE_64BIT_ALIGNMENT
+        || (pParentMT && pParentMT->RequiresAlign8())
+#endif // FEATURE_64BIT_ALIGNMENT
+        ;
+
+    if (bmtLayout->layoutType == EEClassLayoutInfo::LayoutType::Auto)
+    {
+        // Auto layout has been requested.
+        // We never switch away from auto layout, so just go use it right away.
+        if (isAlign8)
+        {
+            GetHalfBakedClass()->SetAlign8Candidate();
+        }
+        HandleAutoLayout(pByValueClassCache);
+        return;
+    }
+
+    _ASSERTE(HasLayout());
+
+    EEClassLayoutInfo* pLayoutInfo = GetLayoutInfo();
+    pLayoutInfo->SetIsBlittable(isBlittable ? TRUE : FALSE);
+    pLayoutInfo->SetHasAutoLayoutField(isAutoLayoutOrHasAutoLayoutField ? TRUE : FALSE);
+    pLayoutInfo->SetIsInt128OrHasInt128Fields(hasInt128Field ? TRUE : FALSE);
+    pLayoutInfo->SetHasExplicitSize(bmtLayout->classSize);
+
+    if (bmtLayout->layoutType == EEClassLayoutInfo::LayoutType::Sequential)
+    {
+        if (hasNonTrivialParent && !pParentMT->IsManagedSequential())
+        {
+            // If the parent type is not Object, ValueType or Sequential, then we need to use Auto layout.
+            bmtLayout->layoutType = EEClassLayoutInfo::LayoutType::Auto;
+        }
+
+        if (hasGCFields)
+        {
+            // If this type has GC fields, we will use Auto layout instead of Sequential layout.
+            bmtLayout->layoutType = EEClassLayoutInfo::LayoutType::Auto;
+        }
+    }
+
+    if (bmtLayout->layoutType == EEClassLayoutInfo::LayoutType::Auto)
+    {
+        if (isAlign8)
+        {
+            GetHalfBakedClass()->SetAlign8Candidate();
+        }
+        HandleAutoLayout(pByValueClassCache);
+        return;
+    }
+
+    // For types with layout we drop any 64-bit alignment requirement if the packing size was less than 8
+    // bytes (this mimics what the native compiler does and ensures we match up calling conventions during
+    // interop).
+    // We don't do this for types that are marked as sequential but end up with auto-layout due to containing pointers,
+    // as auto-layout ignores any Pack directives.
+    if (bmtLayout->packingSize <= 8)
+    {
+        isAlign8 = false;
+    }
```

Comment:
```suggestion
    isAlign8 = !(bmtLayout->packingSize <= 8);
```

Reply:
We don't want to set `isAlign` to true if the `packingSize` would be the only reason we'd set the flag. There needs to be a field the requires it for us to set it (packing can't force wider alignment, only narrower alignment).